hi-ml-azure/src/health_azure/datasets.py

```diff
@@ -104,8 +109,51 @@ def __init__(self,
         self.datastore = datastore
         self.version = version
         self.use_mounting = use_mounting
-        self.target_folder = target_folder
-        self.local_folder = local_folder
+        self.target_folder = Path(target_folder) if target_folder is not None else None
+        self.local_folder = Path(local_folder) if local_folder is not None else None
+
+    def to_input_dataset_local(self, workspace: Optional[Workspace]) -> Tuple[Optional[Path], Optional[MountContext]]:
+        """
+        Return a local path to the dataset when outside of an AzureML run.
+        If local_folder is supplied, then this is assumed to be a local dataset, and this is returned.
+        Otherwise the dataset is mounted or downloaded to either the target folder or a temporary folder and that is
+        returned.
+
+        :param workspace: The AzureML workspace to read from.
+        :return: Pair of optional path to dataset and optional mountcontext.
+        """
+        status = f"Dataset {self.name} will be "
+
+        if self.local_folder is not None:
+            status += f"obtained from local folder {str(self.local_folder)}"
+            logging.info(status)
```

Comment:
we are printing everything here to logging.info. But some of that is maybe important enough to print to the console?